pycm/pycm_util.py

```diff
@@ -322,27 +327,61 @@ def matrix_params_calc(actual_vector, predict_vector, sample_weight):
     :type predict_vector : list
     :param sample_weight : sample weights list
     :type sample_weight : list
+    :param classes: ordered labels of classes
+    :type classes: list
     :return: [classes_list,table,TP,TN,FP,FN]
     """
     [actual_vector, predict_vector] = vector_filter(
         actual_vector, predict_vector)
     if isinstance(sample_weight, numpy.ndarray):
         sample_weight = sample_weight.tolist()
-    classes = set(actual_vector).union(set(predict_vector))
-    if len(classes) == 1:
-        classes.add("~other~")
-    classes = sorted(classes)
-    map_dict = {k: 0 for k in classes}
-    table = {k: map_dict.copy() for k in classes}
+    [actual_vector, predict_vector, classes_list] = classes_filter(
+        actual_vector, predict_vector, classes)
+    map_dict = {k: 0 for k in classes_list}
+    table = {k: map_dict.copy() for k in classes_list}
     weight_vector = [1] * len(actual_vector)
     if isinstance(sample_weight, (list, numpy.ndarray)):
         if len(sample_weight) == len(actual_vector):
             weight_vector = sample_weight
     for index, item in enumerate(actual_vector):
-        table[item][predict_vector[index]] += 1 * weight_vector[index]
-    [classes, table, TP_dict, TN_dict, FP_dict,
+        if item in classes_list and predict_vector[index] in classes_list:
+            table[item][predict_vector[index]] += 1 * weight_vector[index]
+    [_, _, TP_dict, TN_dict, FP_dict,
         FN_dict] = matrix_params_from_table(table)
-    return [classes, table, TP_dict, TN_dict, FP_dict, FN_dict]
+    return [classes_list, table, TP_dict, TN_dict, FP_dict, FN_dict]
+
+
+def classes_filter(actual_vector, predict_vector, classes=None):
+    """
+    Return updated vectors and classes list.
+
+    :param actual_vector: actual values
+    :type actual_vector : list
+    :param predict_vector: predict value
+    :type predict_vector : list
+    :param classes: ordered labels of classes
+    :type classes: list
+    :return: [actual_vector, predict_vector, classes_list]
+    """
+    classes_list = set(actual_vector).union(set(predict_vector))
+    if len(classes_list) == 1:
+        classes_list.add("~other~")
+    classes_list = sorted(classes_list)
+    if isinstance(classes, list):
```

Comment:
Just as a suggestion. Shouldn't other types of iterables such as sets be considered? If not, I think at least a warning should be thrown rather than ignoring it.

Reply:
Good idea!
It would be better if we print a warning message in a case that `classes` is neither a `list` nor `None`
@sadrasabouri ?

Reply:
Yes I think that would be fine.